src/app/pages/storage2/modules/devices/components/devices/devices.component.ts

```diff
@@ -104,12 +114,30 @@ export class DevicesComponent implements OnInit {
     this.selectedParentItem = undefined;
   }
 
-  onRowSelected(dataNodeSelected: DeviceNestedDataNode, event: MouseEvent): void {
-    event.stopPropagation();
-    this.selectedItem = dataNodeSelected;
-    this.selectedParentItem = findInTree(this.treeControl.dataNodes, (dataNode: DeviceNestedDataNode) => {
-      return dataNode.guid === dataNodeSelected.guid;
-    });
+  private listenForRouteChanges(id: string): void {
+    this.selectedItem = undefined;
+    this.selectedParentItem = undefined;
+
+    if (id && this.dataSource?.data) {
+      findInTree(this.dataSource.data, (dataNode) => {
+        if (dataNode.children?.length && dataNode.guid !== id) {
+          const item = dataNode.children.find((child) => child.guid === id);
+          if (item) {
+            this.selectedItem = item;
+            this.selectedParentItem = dataNode;
+            return true;
+          }
+          return false;
+        }
+
+        if (dataNode.guid === id) {
+          this.selectedItem = dataNode;
+          this.selectedParentItem = undefined;
+          return true;
+        }
+        return false;
+      });
+    }
```

Comment:
When we're assigning a value to a `selectedItem` we're missing a method call which will expand all the parent nodes to let the user see the node, which we're  set as selected. This video depicts the bug:

   https://user-images.githubusercontent.com/20611516/180776241-03e42cfe-7c47-4de5-97f0-300d82c92f4b.mp4

  `this.treeControl.expand(node)` looks like suitable method to fix that.

Reply:
Fixed.